telemetry/definitions/commonDefinitions.json

```diff
@@ -1438,6 +1443,11 @@
             "type": "string",
             "description": "User locale. Examples: en-US, en-GB, etc."
         },
+        {
+            "name": "logStreamFilterType",
+            "type": "string",
+            "description": "Type of LogStream filter applied to session"
```

Comment:
We have two different filter parameters:
* logStream: determines from which logStream(s) log events are included
* LogEvent: only includes logEvents that match a specified pattern. 

For LogStream filter, we have multiple "types" of filters. Being one of: 'prefix', 'all', or 'specific'. We want to capture what type of filter is being used in VSCode. 
Conversely, there is no "type" to a logEventFilter. It is always a text pattern, thus why it is set as a boolean.

Are we able to emit the same metric twice in the same `telemetry.run()` and get two data points? Say we have `filterPattern` as the metric name as suggested, and we have two properties that have filters. Would this lock us in to only emitting one `filterPattern` value per request?

This doesn't immediately impact us since we only have one case where we are emitting a "filterType", but I could see this being problematic if we wanted to emit a logStreamFilterType AND a logEventFilterType.

Reply:
If we change the type's name to just `filterType`, and we have multiple `filterType` for a given request, how would that work? 

In our case this isn't much of an issue since the only filterType we have is for LogStreams. But hypothetically if we were also emitting a `filterType` for LogEvents, how could we set both for the same request?

Also if I make this an enum, can I make the accepted values specific to CWL (`logStream_prefix`) or should it be generic (`prefix`)?

Reply:
> But hypothetically if we were also emitting a filterType for LogEvents, how could we set both for the same request?

Would it be on a single metric? 

> if I make this an enum, can I make the accepted values specific to CWL (`logStream_prefix`) or should it be generic (`prefix`)?

Feel free to add any and all useful values to an enum, but starting with generic 'prefix', 'all', 'specific' seems fine. Fields exist on metrics, thus they are inherently "namespaced" by whatever metric they are attached to.

Reply:
Yeah in a single metric. Again this is mainly a hypothetical, but we are currently emitting a single metric per "startLiveTail" request. If we wanted to emit a filterType for two different parameters on that request, that might be problematic.

This isn't a blocker to this PR, so i'm moving forward with renaming to `filterType` with allowedValues: ['all', 'prefix', 'specific']